pkg/nodeagent/controller/operatingsystemconfig/reconciler.go

```diff
@@ -149,6 +150,13 @@ func (r *Reconciler) Reconcile(ctx context.Context, request reconcile.Request) (
 		return reconcile.Result{}, fmt.Errorf("unable to write current OSC to file path %q: %w", lastAppliedOperatingSystemConfigFilePath, err)
 	}
 
+	if mustRestartGardenerNodeAgent {
+		log.Info("Restarting myself (gardener-node-agent unit)")
+		if err := r.DBus.Restart(ctx, r.Recorder, node, nodeagentv1alpha1.UnitName); err != nil {
```

Comment:
Do we want to consider exiting instead as systemd will restart the service anyway?
This would give us a clean cut-off point and proper logs.